packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/DeleteRulesTests.java

```diff
@@ -0,0 +1,231 @@
+    {
+        final FolderModel ruleFolder = dataContent.usingUser(user).usingSite(site).createFolder();
+        final RestRuleModel testRule = createRule(ruleFolder);
+
+        STEP("Attempt delete the rule in non-existing rule set");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingRuleSet(FAKE_NODE_REF).deleteRule(testRule.getId());
+
+        restClient.assertLastError().statusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Try to delete a non-existing rule and get 404.
+     */
+    @Test(groups = {TestGroup.REST_API, TestGroup.RULES, TestGroup.SANITY})
+    public void deleteNonExistingRuleAndGet404()
+    {
+        final FolderModel ruleFolder = dataContent.usingUser(user).usingSite(site).createFolder();
+        STEP("Attempt delete non-existing rule");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet().deleteRule(FAKE_NODE_REF);
+
+        restClient.assertLastError().statusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Try to delete an existing rule passing a wrong but existing folder and get 404.
+     */
+    @Test(groups = {TestGroup.REST_API, TestGroup.RULES, TestGroup.SANITY})
+    public void deleteExistingRuleFromWrongFolderAndGet404()
+    {
+        final FolderModel ruleFolder = dataContent.usingUser(user).usingSite(site).createFolder();
+        final RestRuleModel testRule = createRule(ruleFolder);
+
+        STEP("Create a second folder in the site");
+        final FolderModel anotherFolder = dataContent.usingUser(user).usingSite(site).createFolder();
+
+        STEP("Attempt delete an existing rule from a wrong but existing (second) folder");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(anotherFolder).usingDefaultRuleSet().deleteRule(testRule.getId());
+
+        restClient.assertLastError().statusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Check that a user without write permission on folder cannot delete a rule inside it.
+     */
+    public void deleteSingleRuleWithoutPermissionAndGet403()
+    {
+        STEP("Create a user and use them to create a private site containing a folder with a rule");
+        final UserModel privateUser = dataUser.createRandomTestUser();
+        final SiteModel privateSite = dataSite.usingUser(privateUser).createPrivateRandomSite();
+        final FolderModel privateFolder = dataContent.usingUser(privateUser).usingSite(privateSite).createFolder();
+        final RestRuleModel ruleModel = new RestRuleModel();
+        ruleModel.setName("Private site rule");
+        final RestRuleModel createdRule =
+                restClient.authenticateUser(privateUser).withCoreAPI().usingNode(privateFolder).usingDefaultRuleSet()
+                        .createSingleRule(ruleModel);
+
+        STEP("Try to delete the rule with another user");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(privateFolder).usingDefaultRuleSet().deleteRule(createdRule.getId());
```

Comment:
Let's use a collaborator here and make it a requirement that collaborators can't delete private rules.

Reply:
Not sure if it's the same across acs, but for versioning and uploads, collaborators usually have the same privileges as contributors/managers

Reply:
I assume this comes under "Delete content created by another user" - only Managers have that permission:
![image](https://user-images.githubusercontent.com/11535082/181712626-cf49556e-18d0-4e5b-9a86-26416ebe4463.png)

Reply:
So it would be good to test:
* Consumer cannot create a rule
* Contributor can delete their own rule
* Collaborator cannot delete rule created by another user
* Manager can delete rule created by another user

Reply:
I've added a test for collaborator (forbidden to delete) and there is still is the one that covers successful delete by manager.